### PR TITLE
poo#29413: Fix various RAID & LVM use cases on storage-ng

### PR DESCRIPTION
**Pull request by mnowaksuse**
Validation runs:

* SLES 15 (build 414.13)
    - lvm+RAID1@svirt-hyperv-uefi: http://assam.suse.cz/tests/212
    - lvm+RAID1@svirt-hyperv: http://assam.suse.cz/tests/207
    - lvm+RAID1@svirt-xen-pv: http://assam.suse.cz/tests/209
    - lvm+RAID1@svirt-xen-hvm: http://assam.suse.cz/tests/210
    - lvm+RAID1@uefi: http://assam.suse.cz/tests/248
    - RAID6@64bit: http://assam.suse.cz/tests/223
    - RAID5@64bit: http://assam.suse.cz/tests/222

* SLES 12 SP4 (build 135)
    - RAID6: http://assam.suse.cz/tests/243
    - RAID5: http://assam.suse.cz/tests/242
    - lvm+RAID1: http://assam.suse.cz/tests/246

* Leap 15 (build 87.1)
    - RAID0: http://assam.suse.cz/tests/249

* Tumbleweed (build 20180109)
    - lvm+RAID1: http://assam.suse.cz/tests/247
    - RAID10: http://assam.suse.cz/tests/234
    - RAID1: http://assam.suse.cz/tests/225
    - RAID0: http://assam.suse.cz/tests/237

New needles:
* https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/305
* https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/639
* **NEW**: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/641

### CODE REVIEW

File: tests/installation/partitioning_raid.pm [hunk 1]
```diff
@@ -152,43 +149,60 @@ sub set_lvm {
...
 
 sub modify_uefi_boot_partition {
+    send_key 'tab' if is_storage_ng;
     assert_screen 'partitioning_raid-disk_vdd_with_partitions-selected';
     # fold the drive tree
     send_key 'left';
     assert_screen 'partitioning_raid-hard_disks-unfolded';
     # select first disk
+    if (is_storage_ng) {
+        send_key 'left';
```
Comment (rwx788) [inline tests/installation/partitioning_raid.pm]: Could you please explain what is exactly the problem here?

File: tests/installation/partitioning_raid.pm [hunk 2]
```diff
@@ -152,43 +149,60 @@ sub set_lvm {
...
+    }
     send_key 'right';
     assert_screen 'partitioning_raid-disk_vda_with_partitions-selected';
     # edit first partition
     send_key 'alt-e';
     assert_screen 'partition-format';
-    # format as FAT (first choice)
     send_key 'alt-a';
-    assert_screen 'partitioning_raid-format_fat_UEFI';
+    assert_screen is_storage_ng() ? 'partitioning_raid-format_ext_UEFI' : 'partitioning_raid-format_fat_UEFI';
```
Comment (rwx788) [inline tests/installation/partitioning_raid.pm]: Looks like we have different default values, WDYT if we create needle called partitioning_raid-format_default_UEFI here?

File: tests/installation/partitioning_raid.pm [hunk 3]
```diff
@@ -152,43 +149,60 @@ sub set_lvm {
...
-    # keep default
     send_key $cmd{next};
 
+    assert_screen 'volume-pick-fs-role';
     send_key "alt-o";    # Operating System
     send_key $cmd{next};
+    if (is_storage_ng) {
+        if (check_screen 'volume-empty-mount') {
+            # storage-ng does not suggest mount point for OS role volume,
+            # so mount the volume as root explicitely.
```
Comment (rwx788) [inline tests/installation/partitioning_raid.pm]: s/explicitely/explicitly and please add soft-failure (https://bugzilla.suse.com/show_bug.cgi?id=1073854) as this is a bug. @okurz will also chase you for check_screen with non-zero timeout. Not to waste time, better to do assertion on both tags (volume-empty-mount and volume-mount-as-root), and if match_has_tag returns true for volume-empty-mount add soft-failure and set mount point.

Comment (okurz) [reply tests/installation/partitioning_raid.pm]: Yes I will :)
And also after you are getting used to it, it resembles much more how humans think. Normally people would not just check the screen and be blind for the alternatives showing up at the same time

**Comment by okurz** [comment]
both needle PRs merged

File: tests/installation/partitioning_raid.pm (hunk 2 above)
Comment (mnowaksuse) [reply tests/installation/partitioning_raid.pm]: Sure: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/641.

File: tests/installation/partitioning_raid.pm (hunk 1 above)
Comment (mnowaksuse) [reply tests/installation/partitioning_raid.pm]: I added an inline comment, explaining it. Basically it's a short-cut to get to `vda1`.

See:
* before the code in question: http://assam.suse.cz/tests/248#step/partitioning_raid/155
* after: http://assam.suse.cz/tests/248#step/partitioning_raid/156

Fine with you?

File: tests/installation/partitioning_raid.pm (hunk 3 above)
Comment (mnowaksuse) [reply tests/installation/partitioning_raid.pm]: OK, sorry for the vintage code & non-human coding practices :). Fixed & re-verified: http://assam.suse.cz/tests/263#step/partitioning_raid/198.

**Pull request merged**